packages/react-instantsearch-hooks-server/src/getServerState.tsx

```diff
@@ -57,6 +72,29 @@ export function getServerState(
       return {
         initialResults,
       };
+    })
+    .then((serverState) => {
+      let hasDynamicWidgets = false;
+      if (!refetching) {
+        walkIndex(searchRef.current!.mainIndex, (index) => {
+          hasDynamicWidgets =
+            hasDynamicWidgets ||
+            index
+              .getWidgets()
+              .some((widget) => widget.$$type === 'ais.dynamicWidgets');
+        });
+      }
```

Comment:
not sure if this is the best approach? Maybe a while is better?

Reply:
This is not great because we keep walking the whole render tree after we have the answer we need. A recursive `for of` should do it?

Reply:
for of gets polyfilled with generators (even though this code is technically only for node), but I can rewrite this in a while that doesn't read after detecting a dynamic widget (although these trees never would be *that* big that the number of calls to getWidgets would be superfluous)